source/slang-capture-replay/slang-module.cpp

```diff
@@ -0,0 +1,234 @@
+#include "capture_utility.h"
+#include "slang-module.h"
+
+namespace SlangCapture
+{
+    ModuleCapture::ModuleCapture(slang::IModule* module)
+        : m_actualModule(module)
+    {
+        assert(m_actualModule != nullptr);
+        slangCaptureLog(LogLevel::Verbose, "%s: %p\n", __PRETTY_FUNCTION__, module);
+    }
+
+    ModuleCapture::~ModuleCapture()
+    {
+        m_actualModule->release();
+    }
+
+    ISlangUnknown* ModuleCapture::getInterface(const Guid& guid)
+    {
+        if(guid == ModuleCapture::getTypeGuid())
+            return static_cast<ISlangUnknown*>(this);
+        else
+            return nullptr;
+    }
+
+    SLANG_NO_THROW SlangResult ModuleCapture::findEntryPointByName(
+        char const*     name,
+        slang::IEntryPoint**   outEntryPoint)
+	{
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
+
+        slang::IEntryPoint* pEntryPoint = nullptr;
+        SlangResult res = m_actualModule->findEntryPointByName(name, &pEntryPoint);
+        EntryPointCapture* entryPointCapture = getEntryPointCapture(pEntryPoint);
+        *outEntryPoint = static_cast<slang::IEntryPoint*>(entryPointCapture);
```

Comment:
You may need to check if `outEntryPoint` is a null_ptr, before writing a value to it.
Same for all other parameters that takes double pointer type.

Reply:
By looking at the actual `findEntryPointByName`, we seems expecting crash if the output pointer is nullptr.
Because we don't return any error on that case.

So if I check the nullptr, it will change the behavior, because user program will see the `SLANG_OK` as the return value, but output is nothing.

I think the correct fix will be to fix the actual API, not the capture logic. I will try to fix all the APIs along my way.

Reply:
Fix the nullptr check for output parameter in the actual APIs.